### PR TITLE
Enable vtk write cofunction

### PR DESCRIPTION
**Pull request by Ig-dolci**
# Description

### CODE REVIEW

File: firedrake/output/vtk_output.py [hunk 1]
```diff
@@ -496,8 +502,8 @@ def _write_vtu(self, *functions):
             functions = [Function(V)]
 
         for f in functions:
-            if not isinstance(f, Function):
-                raise ValueError("Can only output Functions or a single mesh, not %r" % type(f))
+            if not isinstance(f, (Function, Cofunction)):
+                raise ValueError("Can only output Functions, Cofunctions or a single mesh, not %r" % type(f))
```
Comment (connorjward) [inline firedrake/output/vtk_output.py:506]:
```suggestion
                raise ValueError(f"Can only output Functions, Cofunctions or a single mesh, not {type(f).__name__}")
```

File: firedrake/output/vtk_output.py [hunk 2]
```diff
@@ -477,16 +477,22 @@ def _prepare_output(self, function, max_elem):
                                     shape=shape)
         else:
             raise ValueError("Unsupported shape %s" % (shape, ))
-        output = Function(V)
+        if isinstance(function, Function):
+            output = Function(V)
+        elif isinstance(function, Cofunction):
```
Comment (connorjward) [inline firedrake/output/vtk_output.py:482]:
```suggestion
        else:
            assert isinstance(function, Cofunction)
```
This is just my personal preference. Means that `function` can't be anything else.

**Review by connorjward — CHANGES_REQUESTED** [review]
Otherwise happy.

File: firedrake/output/vtk_output.py (hunk 2 above)
Comment (Ig-dolci) [reply firedrake/output/vtk_output.py:482]: Okay. I am fine with `assert` here. We have a previous check with an error message raised.

**Pull request merged**